src/minefield.rs

```diff
@@ -144,15 +159,36 @@ impl Board {
     /// Uncovering every non-mine tile is the win condition.
     /// Note that the mine tiles are **not** required to be flagged (looking at you, speedrunners).
     pub fn handle_uncover(&mut self, x: usize, y: usize) {
-        todo!()
+        let tile_idx = Self::coords_to_index(&self.params, x, y);
+        match self.tiles[tile_idx].cover {
+            Cover::Up(_) => {
+                self.tiles[tile_idx].uncover();
+                if self.covered_count > self.params.mines {
+                    self.covered_count -= 1;
+                }
+                match self.tiles[tile_idx].object {
+                    Object::Mine => self.defeat = true,
+                    Object::Blank => self.flood_uncover(x, y),
+                    _ => ()
+                }
+            }
+            Cover::Down => () //OPTIONAL: on uncover additional action when clicking hint
+        }
     }
 
     /// Primary interface for acting on a minefield.
     ///
     /// Corresponds to the action of cycling through
     /// available covered-field marks (the [Mark] type).
     pub fn handle_mark(&mut self, x: usize, y: usize) {
-        todo!()
+        let tile_idx = Self::coords_to_index(&self.params, x, y);
+        if let Cover::Up(mark) = self.tiles[tile_idx].cover {
```

Comment:
This copies `mark` instead of getting a mutable reference to it

```suggestion
        if let Cover::Up(mark) = &mut self.tiles[tile_idx].cover {
```

Reply:
Yet to be implemented